src/main/java/edu/iis/powp/command/factory/CommandFactoryWindowController.java

```diff
@@ -0,0 +1,133 @@
+            }
+        };
+    }
+
+    ActionListener handleSaveButton(DefaultListModel<IPlotterCommand> jListModel, JComboBox complexComboBox,
+            JTextField commandNameField) {
+        return new ActionListener() {
+
+            @Override public void actionPerformed(final ActionEvent e) {
+                String commandName = commandNameField.getText();
+                CommandFactory commandFactory = CommandFactory.create(commandName);
+                Enumeration<IPlotterCommand> elements = jListModel.elements();
+                while (elements.hasMoreElements()) {
+                    IPlotterCommand command = elements.nextElement();
+                    commandFactory.addCommand(command);
+                }
+                ICompoundCommand newCommand = commandFactory.build();
+                commandFactoryWindow.getCommandRegistry().registerComplexCommand(newCommand);
+                complexComboBox.addItem(newCommand);
+                commandNameField.setText("");
+                jListModel.clear();
+                commandFactoryWindow.getDrawPanelController().clearPanel();
+            }
+        };
+    }
+
+    ActionListener handleComplexAddButton(DefaultListModel<IPlotterCommand> jListModel, JComboBox complexComboBox) {
+        return new ActionListener() {
+
+            @Override public void actionPerformed(final ActionEvent e) {
+                ICompoundCommand selectedItem = (ICompoundCommand) complexComboBox.getSelectedItem();
+                jListModel.addElement(selectedItem);
+                selectedItem.execute(commandFactoryWindow.getLineAdapterPlotterDriver());
+            }
+        };
+    }
+
+    ActionListener handleBasicAddButton(DefaultListModel<IPlotterCommand> jListModel,
+            JComboBox<CommandFactoryWindow.ConstructorDecorator> basicComboBox) {
+        return new ActionListener() {
+
+            @Override public void actionPerformed(final ActionEvent e) {
+                JFrame frame = new JFrame();
+                frame.setBounds(100, 100, 200, 100);
+                CommandFactoryWindow.ConstructorDecorator selectedItem = (CommandFactoryWindow.ConstructorDecorator) basicComboBox
+                        .getSelectedItem();
+                Constructor<? extends IPlotterCommand> constructor = selectedItem.getConstructor();
+                frame.setLayout(new GridLayout(constructor.getParameterCount() + 1, 2));
+                Map<Parameter, JTextField> textFields = new LinkedHashMap<Parameter, JTextField>();
+                for (Parameter parameter : constructor.getParameters()) {
+                    JLabel label = new JLabel(parameter.getName());
+                    frame.add(label);
+                    JTextField field = new JTextField();
+                    frame.add(field);
+                    textFields.put(parameter, field);
+                }
+                JButton button = new JButton("Add");
+                button.addActionListener(new ActionListener() {
```

Comment:
FIX TO MERGE: again, embedded ActionListener and controller not separated from GUI. Try at least unnest this ActionListener code.